### PR TITLE
chore: add 'npm run lint:tav'

### PR DESCRIPTION
**Pull request by trentm**
This checks that .ci/tav.json is correct: both that it covers all the
modules in all the .tav.yml files in the repo, and that it never
results in a TAV test matrix greater than the GH Actions 256 limit.
This is part of `npm run lint` which is run by the lint CI workflow.

Refs: https://github.com/elastic/apm-agent-nodejs/issues/3227 (item 3)

---

This also drops the body-parser TAV tests: we don't instrument it, just use it in some tests.

### CODE REVIEW

File: dev-utils/lint-tav-json.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  tavJson.modules.forEach(m => {
+    m.split(',').forEach(moduleName => {
+      moduleNamesFromJson.add(moduleName)
+    })
+  })
+  // console.log('moduleNamesFromJson: ', moduleNamesFromJson)
+
+  // Matrix 256 limit.
+  const matrixSize = tavJson.versions.length * tavJson.modules.length
+  if (matrixSize > 256) {
```
Comment (v1v) [inline dev-utils/lint-tav-json.js:61]: That's pretty neat! If it helps the fix could be refactoring the existing TAV to run in chunks as done with https://github.com/elastic/apm-agent-python/blob/1e38ec53115edc70c36c6485259733a8cde02ed9/.github/workflows/test.yml#L59-L81

**Review by v1v — APPROVED** [review]

File: dev-utils/lint-tav-json.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  )
+  // console.log('tavYmlPaths:', tavYmlPaths)
+  tavYmlPaths.forEach(p => {
+    const tavCfg = yaml.load(fs.readFileSync(p, 'utf8'))
+    Object.keys(tavCfg).forEach(k => {
+      const v = tavCfg[k]
+      moduleNamesFromYaml.add(v.name || k)
+    })
+  })
+  // console.log('moduleNamesFromYaml: ', moduleNamesFromYaml)
```
Comment (david-luna) [inline dev-utils/lint-tav-json.js:47]: nit: do you want to keep these logs?

**Review by david-luna — APPROVED** [review]

File: dev-utils/lint-tav-json.js (hunk 2 above)
Comment (trentm) [reply dev-utils/lint-tav-json.js:47]: Kind of, yes. :)  If I ever have to come back and debug this script, then those would be the first interesting things I'd want to re-enable. However, granted, this is a simple script and I don't expect we'll need to come back and debug. I don't care that strongly either way.

To get waaay off track here: I sometime wonder at code (perhaps a language could have specific features/syntax for this) having ways of indicating points of interest for the debugging process.

Comment (trentm) [reply dev-utils/lint-tav-json.js:47]: I guess the next (better?) step would be to add a logging lib, a command-line option and/or envvar to set the debug level, and make these log.debug statements.

**Pull request merged**